tasks/tools.py

```diff
@@ -49,12 +53,113 @@ def remove_from_do(id):
 
     session.execute("COMMIT")
 
+
+def extract_nulls_from_log(infile, outfile, relevant_fields):
+    STR_NULL = 'contain only NULL values: '
+    STR_ERROR = '[ERROR]'
+    STR_INFO = '[INFO]'
+    STR_EXCEPTIONS = 'exceptions'
+
+    relevant_fields = relevant_fields.split(',')
+    cols = {}
+
+    try:
+        os.remove(outfile)
+    except OSError:
+        pass
+
+    with open(outfile, 'a') as fout:
+        with open(infile) as fin:
+            current_params = {}
+            for line in fin:
+                if STR_ERROR in line:
+                    current_params = {}
+                    try:
+                        vals = {item.split("=")[0].strip(): item.split("=")[1].strip()
+                                for item in line[line.rfind('(') + 1:line.rfind(')')].split(',')}
+                        for key, val in vals.items():
```

Comment:
Since `current_params` has just been emptied, I think this loop is equivalent to:

`current_params = { k: v for k, v in vals.items() if k in relevant_fields }`